rj_gameplay/rj_gameplay/skill/mark.py

```diff
@@ -23,15 +50,40 @@ class IMark(skill.ISkill, ABC):
 """
 class Mark(IMark):
 
-    def __init__(self, mark_heuristic: Callable[[np.array], float]):
-        self.__name__ = 'Mark'
-        self.mark_heuristic = mark_heuristic
-        self.mark_behavior = CostBehavior('Mark', self.mark_heuristic)
+    def __init__(self,
+            robot : rc.Robot = None,
+            target_robot : rc.Robot = None):
+
+        self.__name__ = 'Mark Skill'
+        self.robot = robot
+        self.target_robot = target_robot
+
+        if self.robot is not None:
+            self.move = move.Move(self.robot.id, np.array([0.0, 0.0]), np.array([0.0, 0.0]), None, None)
+        else:
+            self.move = move.Move(None, np.array([0.0, 0.0]), np.array([0.0, 0.0]), None, None)
+
+        self.mark_behavior = ActionBehavior('Mark', self.move)
         self.root = self.mark_behavior
         self.root.setup_with_descendants()
 
-    def tick(self, world_state: rc.WorldState, robot: rc.Robot) -> None:
-        print('marking')
-        # Print for stub
-        self.root.tick_once(robot)
-        # TODO: change so this properly returns the actions intent messages
+    def tick(self, robot: rc.Robot, world_state: rc.WorldState) -> None:
+        self.robot = robot
+
+        # update target point every tick to match movement of ball & target robot
+        if world_state and world_state.ball.visible:
+            if self.target_robot is None:
+                mark_point = get_mark_point(1, world_state)
+            else:
+                mark_point = get_mark_point(self.target_robot.id, world_state)
+
+            if mark_point is None: 
+                return None
```

Comment:
If this fails out, then the outer tactic fails to assign and new roles are generated, right? cc @HussainGynai. Just trying to make sure I understand the dataflow for the new gameplay system.

Reply:
I don't think it should return None. It should probably return an empty list, since skill normally return a list of actions, but this shouldn't be able to get a None point anyway I think.